### PR TITLE
(EAI-403) configurable collection name for EmbeddedContentStore & PageStore

### PR DESCRIPTION
**Pull request by nlarew**
Jira: (EAI-403) configurable collection name for EmbeddedContentStore & PageStore

### CODE REVIEW

File: packages/mongodb-rag-core/src/MongoDbPageStore.ts [hunk 1]
```diff
@@ -14,24 +14,37 @@ export type MongoDbPageStore = DatabaseConnection &
...
       args?: LoadPagesArgs<Filter<PersistedPage>>
     ): Promise<PersistedPage[]>;
+    metadata: {
+      databaseName: string;
+      collectionName: string;
+    };
   };
 
+export type MakeMongoDbPageStoreParams = MakeMongoDbDatabaseConnectionParams & {
+  collectionName?: string;
```
Comment (mongodben) [inline packages/mongodb-rag-core/src/MongoDbPageStore.ts:24]: can you add a tsdoc comment about how default is "pages"

smthn like:

```suggestion
  /**
    @default "pages"
   */
  collectionName?: string;
```

**Review by mongodben — APPROVED** [review]
lgtm, see suggestions about the tsdoc comments

**Comment by mongodben** [comment]
also, pls fix CI issue before merge

File: packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.test.ts [hunk 2]
```diff
@@ -206,4 +209,23 @@ describe("nearest neighbor search", () => {
...
+        connectionUri: MONGODB_CONNECTION_URI,
+        databaseName: store.metadata.databaseName,
+        collectionName: "custom-embedded_content",
+      }
+    );
+
+    expect(storeWithCustomCollectionName.metadata.collectionName).toBe(
+      "custom-embedded_content"
+    );
+    await storeWithCustomCollectionName.drop();
```
Comment (mongodben) [inline packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.test.ts:229]: oopsies!!

Comment (nlarew) [inline packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.test.ts:229]:
```suggestion
```

File: packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.test.ts [hunk 3]
```diff
@@ -118,7 +121,7 @@ describe("nearest neighbor search", () => {
     minScore: 0.9,
   };
 
-  let store: (DatabaseConnection & EmbeddedContentStore) | undefined;
+  let store: MongoDbEmbeddedContentStore | undefined;
   beforeEach(async () => {
     // Need to use real Atlas connection in order to run vector searches
     const databaseName = MONGODB_DATABASE_NAME;
```
Comment (mongodben) [inline packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.test.ts:127]: make this a randomly generated name please

File: packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.test.ts (hunk 2 above)
Comment (mongodben) [reply packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.test.ts]: seems like this is dropping the whole DB when it should be just the collection

File: packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.test.ts (hunk 3 above)
Comment (nlarew) [reply packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.test.ts:127]: unfortunately we can't here - its testing vector search so it needs to point to the db that has the embedded data. Best course is to just not call `db.drop()` (lol) or maybe add a readonly version of the store and use that in these tests

**Pull request merged**